pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1486,30 +1526,31 @@ func TestMergePullSecrets(t *testing.T) {
 			name:             "merged pull secret should be be equal to global pull secret",
 			globalPullSecret: `{"auths": {"registry.svc.ci.okd.org": {"auth": "dXNljlfjldsfSDD"}}}`,
 			mergedPullSecret: `{"auths": {"registry.svc.ci.okd.org": {"auth": "dXNljlfjldsfSDD"}}}`,
-			existingCD: []runtime.Object{
-				func() *hivev1.ClusterDeployment {
-					return getCDWithoutPullSecret()
-				}(),
+			existingObjs: []runtime.Object{
+				getCDWithoutPullSecret(),
+				getHiveDeploymentObj(),
```

Comment:
The clusterDeployment controller has no dependency on the hive deployment resource. It is not necessary to add it.